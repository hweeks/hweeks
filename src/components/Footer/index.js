import React from 'react';
import {
  FooterWrapper,
  FooterLink,
} from './styles';

export default () => (
  <FooterWrapper>
      made with <FooterLink href="https://github.com/hweeks/hweeks">git</FooterLink> in the forest
  </FooterWrapper>
);
