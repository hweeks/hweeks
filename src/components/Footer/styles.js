import styled from 'styled-components';

export const FooterWrapper = styled.footer`
  width: 100%;
  font-size: ${props => props.theme.bodyFontSize}rem;
  line-height: ${props => props.theme.headerFontSize}rem;
  text-align: center;
  border-top: ${props => props.theme.borderDefault};
  border-color: ${props => props.theme.syntax.constant};
`;
export const FooterLink = styled.a`
  font-size: ${props => props.theme.bodyFontSize}rem;
  line-height: ${props => props.theme.headerFontSize}rem;
  text-decoration: none;
  color: inherit;

  &:hover {
    text-decoration: underline;
  }
`;
