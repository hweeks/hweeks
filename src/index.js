import ReactDOM from 'react-dom';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import App from './composite/App';

const load = () => {
  ReactDOM.render(
    <BrowserRouter>
      <App />
    </BrowserRouter>,
    document.querySelector('#home'),
  );
};

if (document.readyState !== 'complete') {
  document.addEventListener('DOMContentLoaded', load);
} else {
  load();
}
