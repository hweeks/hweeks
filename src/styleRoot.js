const rootValues = {
  atom: '0.4',
  headerFontSize: '2.2',
  bodyFontSize: '1.6',
  borderColor: '#666',
  bgColor: '#fff',
  borderDefault: `0.2rem solid #666`,
  breakpoint: '41.5',
}

export default rootValues
