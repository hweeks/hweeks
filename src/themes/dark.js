export const dark = {
  common: {
    accent: '#E6B450',
    bg: '#0A0E14',
    fg: '#B3B1AD',
    ui: '#3D424D',
  },
  syntax: {
    tag: '#39BAE6',
    func: '#FFB454',
    entity: '#59C2FF',
    string: '#C2D94C',
    regexp: '#95E6CB',
    markup: '#F07178',
    keyword: '#FF8F40',
    special: '#E6B673',
    comment: '#626A73',
    constant: '#FFEE99',
    operator: '#F29668',
    error: '#FF3333',
  },
  ui: {
    line: '#01060E',
    panel: {
      bg: '#0D1016',
      shadow: '#00010A',
      border: '#000000',
    },
    gutter: {
      normal: '#3D424D99',
      active: '#3D424DE6',
    },
    selection: {
      bg: '#273747',
      inactive: '#1F2C39',
      border: '#304357',
    },
    guide: {
      active: '#3D424DB3',
      normal: '#3D424D66',
    },
  },
  vcs: {
    added: '#91B362',
    modified: '#6994BF',
    removed: '#D96C75',
  },
};
