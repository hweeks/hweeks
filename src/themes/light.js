export const light = {
  common: {
    accent: '#FF9940',
    bg: '#FAFAFA',
    fg: '#6C7680',
    ui: '#959DA6',
  },
  syntax: {
    tag: '#55B4D4',
    func: '#F2AE49',
    entity: '#399EE6',
    string: '#86B300',
    regexp: '#4CBF99',
    markup: '#F07171',
    keyword: '#FA8D3E',
    special: '#E6BA7E',
    comment: '#ABB0B6',
    constant: '#A37ACC',
    operator: '#ED9366',
    error: '#F51818',
  },
  ui: {
    line: '#959DA61A',
    panel: {
      bg: '#FFFFFF',
      shadow: '#56606940',
      border: '#F0F0F0',
    },
    gutter: {
      normal: '#959DA666',
      active: '#959DA6CC',
    },
    selection: {
      bg: '#EDF0F5',
      inactive: '#F2F4F7',
      border: '#E5EBF2',
    },
    guide: {
      active: '#959DA6B3',
      normal: '#959DA64D',
    },
  },
  vcs: {
    added: '#99BF4D',
    modified: '#709ECC',
    removed: '#F27983',
  },
};
