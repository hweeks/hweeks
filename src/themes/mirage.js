export const mirage = {
  common: {
    accent: '#FFCC66',
    bg: '#1F2430',
    fg: '#CBCCC6',
    ui: '#707A8C',
  },
  syntax: {
    tag: '#5CCFE6',
    func: '#FFD580',
    entity: '#73D0FF',
    string: '#BAE67E',
    regexp: '#95E6CB',
    markup: '#F28779',
    keyword: '#FFA759',
    special: '#FFE6B3',
    comment: '#5C6773',
    constant: '#D4BFFF',
    operator: '#F29E74',
    error: '#FF3333',
  },
  ui: {
    line: '#191E2A',
    panel: {
      bg: '#232834',
      shadow: '#141925',
      border: '#101521',
    },
    gutter: {
      normal: '#707A8C66',
      active: '#707A8CCC',
    },
    selection: {
      bg: '#34455A',
      inactive: '#2D3B4D',
      border: '#3C526A',
    },
    guide: {
      active: '#707A8CB3',
      normal: '#707A8C4D',
    },
  },
  vcs: {
    added: '#A6CC70',
    modified: '#77A8D9',
    removed: '#F27983',
  },
};
