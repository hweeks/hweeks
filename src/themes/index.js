import { light as lTheme } from './light';
import { mirage as mTheme } from './mirage';
import { dark as dTheme } from './dark';

const rootValues = {
  atom: '0.4',
  headerFontSize: '2.2',
  bodyFontSize: '1.6',
  borderColor: '#666',
  bgColor: '#fff',
  borderDefault: '0.2rem solid #666',
  breakpoint: '41.5',
};

export const light = { ...rootValues, ...lTheme };

export const dark = { ...rootValues, ...dTheme };

export const mirage = { ...rootValues, ...mTheme };
