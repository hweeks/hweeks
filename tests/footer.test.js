import React from 'react';
import renderer from 'react-test-renderer';
import Footer from '../src/components/Footer';
import { ThemeProvider } from 'styled-components'
import { light } from '../src/themes'

test('Footer actually renders', () => {
  const component = renderer.create(
    <ThemeProvider theme={light}>
      <Footer />
    </ThemeProvider>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
