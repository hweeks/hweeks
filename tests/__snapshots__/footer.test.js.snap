// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer actually renders 1`] = `
<footer
  className="sc-bdVaJa hexMQ"
>
  made with 
  <a
    className="sc-bwzfXH hMjEnB"
    href="https://github.com/hweeks/hweeks"
  >
    git
  </a>
   in the forest
</footer>
`;
